reflex/components/graphing/recharts/utils.py

```diff
@@ -0,0 +1,389 @@
+    #y_series = melted_df[new_y_column_name]
+
+    # if (
+    #     y_series.dtype == "object"
+    #     and "mixed" in infer_dtype(y_series)
+    #     and len(y_series.unique()) > 100
+    # ):
+    #     raise ValueError(
+    #         "The columns used for rendering the chart contain too many values with mixed types. Please select the columns manually via the y parameter."
+    #     )
+
+    #return fixed_df
+    return melted_df
+
+
+def _maybe_melt(
+    df: pd.DataFrame,
+    x_column: str | None,
+    y_column_list: List[str],
+) -> Tuple[pd.DataFrame, str | None, str | None]:
+    """If multiple columns are set for y, melt the dataframe into long format."""
+    y_column: str | None
+
+    if len(y_column_list) == 0:
+        y_column = None
+
+    y_column = "value--p5bJXXpQgv"
+
+    columns_to_leave_alone = [x_column]
+
+    df = _melt_data(
+        df=df,
+        columns_to_leave_alone=columns_to_leave_alone,
+        columns_to_melt=y_column_list,
+        new_y_column_name=y_column,
+    )
+
+    return df, y_column
+
+
+def prep_data(
+    df: pd.DataFrame,
+    x_column: str | None,
+    y_column_list: List[str],
+) -> Tuple[pd.DataFrame, str | None, str | None, str | None, str | None]:
+    """Prepares the data for charting. This is also used in add_rows.
+
+    Returns the prepared dataframe and the new names of the x column (taking the index reset into
+    consideration) and y, color, and size columns.
+    """
+
+    # If y is provided, but x is not, we'll use the index as x.
+    # So we need to pull the index into its own column.
+    x_column = _maybe_reset_index_in_place(df, x_column, y_column_list)
+
+    # Drop columns we're not using.
+    selected_data = _drop_unused_columns(df, x_column, *y_column_list)
+
+    # Make sure all columns have string names.
+    x_column, y_column_list = _convert_col_names_to_str_in_place(selected_data, x_column, y_column_list)
+
+    # Maybe melt data from wide format into long format.
+    melted_data, y_column = _maybe_melt(selected_data, x_column, y_column_list)
+
+    # Return the data, but also the new names to use for x, y, and color.
+    return melted_data, x_column, y_column
+
+
+def prepare_y_and_color(y, color):
+    
+    # Set default color if color is None
+    default_color = "#8884d8"
+    if color is None:
+        color = [default_color] * len(y)
+
+    if isinstance(y, str):
+        y = [y]
+
+    if isinstance(color, str):
+        color = [color]
+
+    # If both are lists and have the same length, zip them together
+    if isinstance(y, list) and isinstance(color, list) and len(y) == len(color):
+        return zip(y, color)
+
+    # If lists are of different lengths, raise an error
+    else:
+        raise ValueError("y and color must be of the same length")
+    
+
+
+def bar_chart_high_api(
+        data: pd.DataFrame | np.ndarray | Dict[str, List[str | int]] | List[Dict[str, Any]], 
```

Comment:
the fact that this doesn't work with state seems like a big incongruity in the API.

it seems like it mostly could work with state as long as the data is in the right format, except for `convert_anything_to_dataframe` really wanting a python value, not a Var